pyat/at/physics/frequency_maps.py

```diff
@@ -0,0 +1,283 @@
+                        ):
+    """
+    This function calculates the norm of the transverse tune variation per turn
+    for a particle tracked along a ring with a set of offsets in the initial
+    coordinates.
+
+    It returns a numpy array containing 7 columns,
+        [xoffset, yoffset, nux, nuy, dnux, dnuy,
+            log10( sqrt(dnux*dnux + dnuy*dnuy)/tns )]
+    for every tracked particle that survives 2*tns turns.
+    Particles lost before 2*tns turns are ignored.
+    The log10 tune variation is limited to the interval from -10 to -2;
+    values above or below are set to the closer limit.
+
+    The transverse offsets are given inside a rectangular coordinate window
+        coords=[xmin,xmax,ymin,ymax]
+    in millimeters, where the window is divided in n steps=[xsteps,ysteps].
+    For each yoffset, particle tracking over the whole set of xoffsets is done
+    in parallel, therefore, a number of cpus (ncpu) above the number of xsteps
+    will not reduce the calculation time.
+
+    The frequency analysis uses a library that is not parallelized.
+
+    The closed orbit (co) could be calculated and added to the
+    initial particle offset of every particle by setting co=True.
+    Additionally, a six by one numpy array (add_offset6D) could be used to
+    arbitrarily offset the initial coordinates of every particle.
+
+    A dictionary with particle losses is saved for every vertical offset.
+    See the patpass documentation.
+
+    Parameters :
+        ring:     a valid pyat ring
+    Optional:
+        coords:   default [-10,10,-10,10] in mm
+        steps:    default [100,100]
+        scale:    default 'linear'; or 'non-linear'
+        turns:    default 512
+        ncpu:     default 30; max. number of processors
+                                in parallel tracking patpass
+        co:       default true
+        add_offset6D: default numpy.zeros((6,1))
+        verbose:  prints additional info
+        lossmap:  default false
+    Returns:
+        xy_nuxy_lognudiff_array: numpy array with columns
+                    [xcoor, ycoor, nux, ny, dnux, dnuy,
+                        log10(sqrt(sum(dnu**2)/turns)) ]
+        loss_map_array : experimental format.
+                    if loss_map is True, it returns the losses dictionary
+                    provided by patpass per every vertical offset.
+                    if loss_map is False, it returs a one-element list.
+
+    WARNING : points with NaN tracking results or non-defined frequency
+              in x,y are ignored.
+    WARNING : if co flag is used, the closed orbit is added using
+                at.physics.find_orbit
+              however, it is not returned by this function.
+    WARNING : loss map format is experimental
+    """
+
+    if co:
+        # closed orbit values are not returned. It seems not necessary here
+        addco = find_orbit(ring)
+        print(f'Closed orbit:\t{addco}')
+    else:
+        addco = [numpy.zeros((6, 1))]
```

Comment:
The usual way to handle closed orbit in PyAT is to add a keyword argument `orbit=None`. Then, here you would have:
```python
if orbit is None:
    # closed orbit values are not returned. It seems not necessary here
    addco = find_orbit(ring)
    print(f'Closed orbit:\t{addco}')
else:
    addco = orbit
```
You can then remove the `co` flag. And even simpler, use directly `orbit` instead of `addco` later.

Reply:
@lfarv 
This change has been implemented in commit @oscarxblanco
[removes co flag, adds orbit](https://github.com/atcollab/at/pull/556/commits/5998db4ec58873ab9b3226c61a66522ccbca9a17)